.github/workflows/ci.yml

```diff
@@ -0,0 +1,26 @@
+name: CI Check
+on: [push, pull_request, workflow_dispatch]
+jobs:
+  clean-and-test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ['3.10', '3.11']
+    steps:
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Getting repository
+      uses: actions/checkout@v4
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+    - name: Install pytest and package
+      run: |
+        pip install pytest
+        pip install -e .
+    - name: Execute PyTest
+      run: pytest 
```

Comment:
To be on the safe side, I would run `python -m pytest` and you can include options like `-svv` to make any failures more apparent and understandable.